ocaml/xapi/create_misc.ml

```diff
@@ -196,9 +201,12 @@ and ensure_domain_zero_record ~__context (host_info: host_info): [`VM] Ref.t =
   let (domain_zero_ref, found) =
     try       ref_lookup (), true
     with _ -> ref_create (), false in
-  if found
-  then update_domain_zero_record ~__context ~domain_zero_ref host_info
-  else create_domain_zero_record ~__context ~domain_zero_ref host_info;
+  if found then begin
+    update_domain_zero_record ~__context ~domain_zero_ref host_info;
+    update_domain_zero_metrics_record ~__context ~domain_zero_ref
```

Comment:
Let's add this to `ensure_domain_zero_metrics_record` instead to avoid more 🍝

Reply:
Done.